tests/src/test/java/org/eclipse/hono/tests/amqp/CommandAndControlAmqpIT.java

```diff
@@ -849,6 +849,45 @@ public void testSendCommandFailsForCommandNotAcknowledgedByDevice(
         }
     }
 
+    /**
+     * Verifies that the adapter immediately closes the command receiver link of a gateway wanting
+     * to receive commands of a device that doesn't have the gateway in its via-gateways.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    @Timeout(timeUnit = TimeUnit.SECONDS, value = 20)
+    public void testCommandReceiverCreationFailsForDeviceNotInViaGateways(
+            final VertxTestContext ctx) {
+
+        final AmqpCommandEndpointConfiguration endpointConfig = new AmqpCommandEndpointConfiguration(
+                SubscriberRole.GATEWAY_FOR_SINGLE_DEVICE);
+
+        final String otherDeviceId = helper.getRandomDeviceId(tenantId);
+        helper.registry
+                .addDeviceToTenant(tenantId, deviceId, password)
+                .compose(ok -> helper.registry.addDeviceToTenant(tenantId, otherDeviceId, password))
+                .compose(ok -> connectToAdapter(IntegrationTestSupport.getUsername(deviceId, tenantId), password))
+                .compose(conAck -> {
+                    final Promise<Void> result = Promise.promise();
+                    context.runOnContext(go -> {
+                        final ProtonReceiver recv = connection
+                                .createReceiver(endpointConfig.getSubscriptionAddress(tenantId, otherDeviceId));
+                        recv.setQoS(ProtonQoS.AT_LEAST_ONCE);
+                        recv.openHandler(openResult -> {
+                            ctx.verify(() -> {
+                                assertThat(openResult.failed()).isTrue();
```

Comment:
I am not sure if the open handler will be invoked with a failed AsyncResult in any case. However, the closeHandler and/or the detachHandler should contain an error that we can assert.

Reply:
Yes, I think in theory the open handler could succeed with the receiver link remote source being null. I'm not sure whether that can actually happen during a somehow delayed test run.
In any case, I've adapted the tests here accordingly.